src/com/trilead/ssh2/transport/TransportManager.java

```diff
@@ -122,6 +122,7 @@ public void run()
 		}
 	}
 
+	String sourceAddress;
```

Comment:
Could this be made private and final with a getter method if external access is required? Non-encapsulated fields are causing us issues on other changes to this code-base, so I'd prefer we don't introduce more.